runelite-client/src/main/java/net/runelite/client/plugins/ExternalPf4jPluginManager.java

```diff
@@ -0,0 +1,370 @@
+package net.runelite.client.plugins;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.client.RuneLite;
+import net.runelite.client.RuneLiteProperties;
+import org.pf4j.BasePluginLoader;
+import org.pf4j.CompoundPluginLoader;
+import org.pf4j.CompoundPluginRepository;
+import org.pf4j.DefaultPluginManager;
+import org.pf4j.DependencyResolver;
+import org.pf4j.DevelopmentPluginRepository;
+import org.pf4j.JarPluginLoader;
+import org.pf4j.JarPluginRepository;
+import org.pf4j.ManifestPluginDescriptorFinder;
+import org.pf4j.PluginAlreadyLoadedException;
+import org.pf4j.PluginDescriptorFinder;
+import org.pf4j.PluginLoader;
+import org.pf4j.PluginRepository;
+import org.pf4j.PluginRuntimeException;
+import org.pf4j.PluginState;
+import org.pf4j.PluginStateEvent;
+import org.pf4j.PluginWrapper;
+import org.pf4j.RuntimeMode;
+
+/**
+ * A pf4j {@link DefaultPluginManager} which can load openosrs external plugins, either from jar files in a directory,
+ * or from locally compiled classes depending on whether {@link ExternalPluginManager#isDevelopmentMode()} is true
+ */
+@Slf4j
+class ExternalPf4jPluginManager extends DefaultPluginManager
+{
+	private final ExternalPluginManager externalPluginManager;
+
+	public ExternalPf4jPluginManager(ExternalPluginManager externalPluginManager)
+	{
+		super(RuneLiteProperties.getPluginPath() != null ? Paths.get(RuneLiteProperties.getPluginPath())
+			: RuneLite.EXTERNALPLUGIN_DIR.toPath());
+		this.externalPluginManager = externalPluginManager;
+	}
+
+	@Override
+	protected PluginDescriptorFinder createPluginDescriptorFinder()
+	{
+		return new ManifestPluginDescriptorFinder()
+		{
+			protected Path getManifestPath(Path pluginPath)
+			{
+				if (isDevelopment())
+				{
+					// The superclass performs a find, which is slow in development mode since we're pointing
+					// at a sources directory, which can have a lot of files. The external plugin template
+					// will always output the manifest at the following location, so we can hardcode this path.
+					return pluginPath.resolve(ExternalPluginManager.DEVELOPMENT_MANIFEST_PATH);
+				}
+
+				return super.getManifestPath(pluginPath);
+			}
+		};
+	}
+
+	@Override
+	protected PluginRepository createPluginRepository()
+	{
+		CompoundPluginRepository compoundPluginRepository = new CompoundPluginRepository();
+
+		if (isNotDevelopment())
+		{
+			JarPluginRepository jarPluginRepository = new JarPluginRepository(getPluginsRoot())
+			{
+				/**
+				 * Identical to {@link JarPluginRepository#getPluginPaths()} but does not sort by modtime
```

Comment:
curious: why? this was already the case before i touched it

Reply:
Recap from discord:

You're right, tested with the override removed. It doesn't error and it can be removed completely